test/disable-goosig-test.js

```diff
@@ -0,0 +1,159 @@
+
+      assert.strictEqual(block.txs[0].inputs.length, 2);
+      assert(await node.chain.add(block));
+    });
+  });
+
+  describe('After Disable', () => {
+    before(async () => {
+      node = newNode();
+      await node.open();
+    });
+
+    after(async () => {
+      await node.close();
+      node.network.airdrop = networkAirdrop;
+    });
+
+    it('should reject GooSig based airdrop in mempool after', async () => {
+      const {disableGoosigHeight} = node.network.airdrop;
+      while (node.chain.height < disableGoosigHeight) {
+        const block = await node.miner.mineBlock();
+        assert(await node.chain.add(block));
+      }
+
+      assert.strictEqual(node.chain.height, disableGoosigHeight);
+      await assert.rejects(node.mempool.addAirdrop(proof),
+        {reason: 'bad-airdrop-disabled'});
+    });
+
+    it('should reject GooSig based airdrop in block after', async () => {
+      const tip = node.chain.tip;
+      const version = await node.chain.computeBlockVersion(tip);
+      const mtp = await node.chain.getMedianTime(tip);
+      const time = Math.max(node.network.now(), mtp + 1);
+
+      const target = await node.chain.getTarget(time, tip);
+      const root = node.chain.db.treeRoot();
+
+      const template = new BlockTemplate({
+        prevBlock: tip.hash,
+        treeRoot: root,
+        reservedRoot: consensus.ZERO_HASH,
+        height: tip.height + 1,
+        version: version,
+        time: time,
+        bits: target,
+        mtp: mtp
+      });
+
+      template.addAirdrop(proof);
+      template.refresh();
+
+      let block = template.toBlock();
+
+      let nonce = 0;
+      while(!block.verifyPOW()) {
+        const proof = template.getProof(nonce++, time, consensus.ZERO_NONCE, consensus.ZERO_HASH);
+        block = template.commit(proof);
+      }
+
+      await assert.rejects(node.chain.add(block),
+        {reason: 'bad-airdrop-disabled'});
+    });
+  });
+
+  describe('Clear Mempool', () => {
+    before(async () => {
+      node = newNode();
+      await node.open();
+
+      await node.mempool.addAirdrop(proof);
+    });
+
+    after(async () => {
+      await node.close();
+      node.network.airdrop = networkAirdrop;
+    });
+
+    it('should remove GooSig based airdrop', async () => {
+      assert.strictEqual(node.mempool.airdrops.size, 1);
+      assert(node.mempool.has(proof.hash()));
+
+      const {disableGoosigHeight} = node.network.airdrop;
+      while (node.chain.height < disableGoosigHeight) {
+        const block = await node.miner.mineBlock();
+        assert(await node.chain.add(block));
+      }
+
+      assert.strictEqual(node.mempool.has(proof.hash()), false);
+      assert.strictEqual(node.mempool.airdrops.size, 0);
+    });
+  });
+});
+
+/**
+ * Create a new FullNode with GooSig
+ * disabled at a specific height.
+ */
+
+function newNode() {
+  const node = new FullNode({
+    memory: true,
+    network: 'regtest'
+  });
+
+  networkAirdrop = node.network.airdrop;
```

Comment:
This line is not resulting in its intended behavior. Mutating `node.network.airdrop` below will mutate `networkAirdrop`. I think we should remove this line and reset the `network.airdrop` properties within the `after` blocks. I'd also like to suggest renaming this function `createNode` (to follow convention across the test suite) and move it to the top of the file.

Reply:
Zippy's test handles this situation in a clear way: https://github.com/handshake-org/hsd/pull/332/files#diff-0bc7309930b0155fa695605f7786d072R13